mantis-runtime-loader/src/main/java/io/mantisrx/runtime/loader/RuntimeTask.java

```diff
@@ -33,4 +33,9 @@ void initialize(
         UserCodeClassLoader userCodeClassLoader);
 
     String getWorkerId();
+
```

Comment:
I would prefer we do not add specific metrics/metadata into these entry-interfaces. can you elaborate on why it's on the runtimeTask for metrics instead of using regular metrics collector tagging? I feel like there are existing metrics path to do this without adding into this interface.

Reply:
@Andyz26 thanks for the review! 

> can you elaborate on why it's on the runtimeTask for metrics instead of using regular metrics collector tagging? 
The job metadata is only available as part of the job submission, not the actual running job itself. From my understanding, we need to hook into the task/job submission to get information like version. Right now, we have written our own AgentMain class, and we can hook into the TaskExecutor `onTaskStarting` to add custom job labels, like the scaffolding that exists here: https://github.com/Netflix/mantis/blob/c1c7f34a1cdf2fea6a84575eb6f6b94eb3dc71a7/mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/AgentV2Main.java#L59 . 

Given the context above, do you have a recommendation for a better place to make this change? Happy to move things around, it's possible I've missed a better metrics path when going through the code.

Reply:
imo since agent to inner task is not always 1 to 1 it would be more consistent to construct such metrics within the worker runtime where you have access to things like MetricsRegistry (I don't know how your current metrics integration is done into with this one).

Reply:
got it, thanks for the feedback! are you opposed to adding `jobVersion` to the `ExecuteStageRequest`? I think we'll need it to capture the `jobVersion` on the worker